src/App.tsx

```diff
@@ -15,33 +16,106 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+export const App: React.FC = () => {
+  const [isStarted, start] = useState(false);
+  const [isReversed, reverse] = useState(false);
+  const [isSortedByAlph, sortByAlph] = useState(false);
+  const [isSortedByLength, sortByLength] = useState(false);
+  const [minLength, filterByLength] = useState(1);
+  const goods = goodsFromServer.filter(good => good.length >= minLength);
+
+  if (isSortedByAlph) {
+    goods.sort((a, b) => a.localeCompare(b));
+  }
+
+  if (isSortedByLength) {
+    goods.sort((a, b) => a.length - b.length);
+  }
+
+  if (isReversed) {
+    goods.reverse();
+  }
+
+  return (
+    <div className="App">
+      <h1 className="title">Goods</h1>
+      {isStarted ? (
+        <>
+          <GoodsList goods={goods} />
+
+          <div className="select is-success">
+            <select
+              name="length"
+              value={minLength}
+              onChange={(event) => filterByLength(+event.target.value)}
+            >
+              {[...new Array(10)].map((_, i) => (
+                <option value={i + 1}>{i + 1}</option>
+              ))}
+            </select>
+          </div>
+
+          <button
+            type="button"
+            onClick={() => {
+              sortByLength(false);
+              sortByAlph(!isSortedByAlph);
+            }}
```

Comment:
move this logic to function-handlers, it's not a common practice to embed code directly